src/userEvent/utils/warnAboutRealTimers.ts

```diff
@@ -0,0 +1,6 @@
+export const warnAboutRealTimers = () => {
+  // eslint-disable-next-line no-console
+  console.warn(`It is not recommended to use userEvent without using fake timers
```

Comment:
I think it's good idea to warn in case using real timers would actually cause delays, e.g. when wait would be 50 ms or longer.

As far as I checked the User Event's code, their default wait is basically 0 (as in `setTimeout(..., 0)`, so there is only pause to process promises and events): https://github.com/testing-library/user-event/blob/main/src/setup/setup.ts#L24

Reply:
Hi! @pierrezimmermannbam @mdjastrzebski - sorry to dredge this up, i've just got a question on this as i'm not sure I understand.

My understanding is that fake timers are a good idea when timers are being used (setTimeout, setInterval, clearTimeout, clearInterval). 

By "some events" do you mean long press only?  if so could we then just show this warning if long press is used?

If not, what timer is happening? and do you recommend we use fake timers on every single test where there we need to press on something?

thanks!

Reply:
Hello @henbrand ! Actually there are timers involved in every press events. This is because in react native, the `onPressOut` callback can only be called at least 130ms after the `onPressIn`. We wait for the `onPressOut` to be called before resolving `userEvent.press` because it's closer to what the user experiences and makes writting tests easier so if you're not using fake timers then every press would last at least 130ms. 

And yes this is only a problem for press events, but this is due to how react native jest preset is made, pressables are not mocked at the same level than texts or textinputs so the mechanism that should be the same is not in practice. 

Regarding fake timers more generally I consider it a good practice to have them enabled by default in your whole test suite because it brings a lot of benefits (predictability, faster tests, control over timers) and I don't see any drawback to having them always enabled. In my experience they're not used by default because it may require a bit of work on the test config to make them work and there have been issues with RNTL before but now these issues have been fixed and it should not be that hard to activate them on a whole test suite so i highly recommend doing so

Hope this helps, let me know if you have any further questions!